multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/spi/jpa/MultiuserJpaWorkspaceDao.java

```diff
@@ -311,7 +316,35 @@ public void onCascadeEvent(BeforeAccountRemovedEvent event) throws Exception {
               (maxItems, skipCount) ->
                   workspaceManager.getByNamespace(
                       event.getAccount().getName(), false, maxItems, skipCount))) {
-        workspaceManager.removeWorkspace(workspace.getId());
+        // Will try to stop running/starting workspaces.
+        // At the moment is impossible to force stop for  STOPPING workspace
+        if (RUNNING.equals(workspace.getStatus()) || STARTING.equals(workspace.getStatus())) {
+          String owner = workspace.getRuntime().getOwner();
+          if (owner.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
+            workspaceManager.stopWorkspace(
+                workspace.getId(), of(REMOVE_WORKSPACE_AFTER_STOP, "true"));
+          } else {
+            tryStopWorkspaceWithSA(workspace);
+          }
+        } else {
+          workspaceManager.removeWorkspace(workspace.getId());
+        }
+      }
```

Comment:
I still don't know how the made changes could help avoid 500 error. 
We may improve UX a bit if we after one loop to remove or stop with remove workspace - will check if there are no more workspaces. If there are any STOPPING - throw exception `409 - The user has stopping workspaces. Try again`

Reply:
we discussed with @vparfonov this part and indeed if there are "STOPPING" workspaces we should encourage to do the retry explicitly